arangod/Statistics/StatisticsFeature.cpp

```diff
@@ -339,6 +349,76 @@ std::map<std::string, std::vector<std::string>> statStrings{
     "Minimum number of concurrent V8 contexts"}},
 };
 
+// Connect legacy statistics with metrics definitions for automatic checks
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+using StatBuilder =
+    std::unordered_map<std::string_view, std::unique_ptr<metrics::Builder const> const>;
+auto makeStatBuilder(std::initializer_list<std::pair<std::string_view, metrics::Builder const* const>> initList)
+    -> StatBuilder {
+  auto unomap = StatBuilder{};
+  unomap.reserve(initList.size());
+  for (auto const& it : initList) {
+    unomap.emplace(it.first, it.second);
+  }
+  return unomap;
+}
+auto const statBuilder = makeStatBuilder({
+    {"bytesReceived", new arangodb_client_connection_statistics_bytes_received()},
```

Comment:
All of these initializations are done when static variables are initialized.
Can this lead to some potential init order fiasco in case the to-be-created metrics depend on other static variables in a different TU?

Reply:
No, everything here is fully local. This is only a temporary check to ensure everything is consistent. Longterm this should be cleaned up anyway so that this check is no longer necessary.